File: Structural/Bridge/Bridge.ts
class RemoteControl {
  public device: Device;

  constructor(device: Device) {
    this.device = device;
  }

  public togglePower() {
    if (this.device.isEnabled) {
      this.device.disable();
    } else {
      this.device.enable();
    }
  }
}

class AdvanceRemoteControl extends RemoteControl {
  public device: Device;

  constructor(device: Device) {
    super(device);
    this.device = device;
  }
  public mute() {
    super.device.setVolume(0);
  }
}

interface Device {
  disable(): void;
  enable(): void;
  setVolume(arg: number): void;
}

class Device implements Device {
  private isEnabled: boolean = false;

  public disable(): void {
    console.log('hello world');
  }

  public enable(): void {
    console.log('hello world');
  }
}

class Television extends Device {
  constructor() {
    super();
  }

  public disable(): void {
    console.log('hello world');
  }
  public enable(): void {
    console.log('hello world');
  }

  public setVolume(arg: number = 0): void {
    console.log('hello world');
  }
}

File: Structural/Bridge/Bridge.py
from __future__ import annotations
from abc import ABCMeta, abstractmethod


class RemoteControl:
    device: Device

    def __init__(self, device: Device) -> None:
        self.device = device

    def toggle_power(self) -> None:
        if self.device.isEnabled:
            self.device.disable()
        else:
            self.device.enable()


class AdvanceRemoteControl(RemoteControl):
    def mute() -> None:
        self.device.setVolume(0)


class Device(metaclass=ABCMeta):
    isEnabled = False

    @abstractmethod
    def disable(self) -> None:
        pass

    @abstractmethod
    def enable(self) -> None:
        pass


class Television(Device):

    isEnabled = False

    def disable(self) -> None:
        print("disabled")

    def enable(self) -> None:
        print("enabled")


device = Television()
remote = RemoteControl(device)

remote.toggle_power()
